### PR TITLE
Auto-discover EventSrv

### PR DESCRIPTION
**Pull request by reset**
General improvements for communicating with an EventSrv

* Auto-discover EventSrv from `--events` flag
* Refactor of EventSrv protocols into their own crate

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/sup/src/manager/events.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        let mut event = EventEnvelope::new();
+        event.set_field_type(EventEnvelope_Type::ProtoBuf);
+        event.set_payload(payload_buf);
+        event.set_member_id(member.member_id.clone());
+        event.set_service(PRODUCT.to_string());
+        self.tx.send(Command::SendEvent(event)).unwrap();
+    }
+
+    pub fn try_connect(&self, census: &CensusRing) {
+        if let Some(cg) = census.census_group_for(&self.group) {
```
Comment (fnichol) [inline components/sup/src/manager/events.rs]: That's super cute!

File: components/sup/src/manager/mod.rs [hunk 2]
```diff
@@ -483,50 +481,10 @@ impl Manager {
...
-                        let _ = client.send(ee);
-                        Ok(())
-                    }
-                    Err(e) => return Err(e),
-                }
-            })
-            .expect("unable to start sup-eventsrv thread");
-
+        let events = match self.events_group {
+            Some(ref evg) => Some(events::EventsMgr::start(evg.clone())),
```
Comment (fnichol) [inline components/sup/src/manager/mod.rs]: Nice!

**Comment by reset** [comment]
Gonna skip sentinels on this PR to speed things up a bit!

**Pull request merged**